### PR TITLE
Automatically add missing uuids while loading station lists

### LINKED ISSUE #529: V.0.74-V.0.76: Message "Cannot connect to server" when trying to open a radio station (stream)
As described in the issues title.
With the previous versions, eventing worked well.
Currently switched back to version 0.73.

### PR DESCRIPTION
**Pull request by morckx**
Fixes #529, #516 and #549

### CODE REVIEW

**Comment by morckx** [comment]
This is somewhere in between the alternative ways A) to make the API also accept IDs as UUIDs and B) a full-fleshed settings migration procedure.

**Comment by zo-shin** [comment]
The toast message should be made translatable.

**Comment by morckx** [comment]
> The toast message should be made translatable.

Yes, I'll probably just delete it. It's really not interesting to any user.

**Comment by morckx** [comment]
Should be pretty fail-proof, now.

File: app/src/main/java/net/programmierecke/radiodroid2/data/DataRadioStation.java [hunk 1]
```diff
@@ -75,6 +79,14 @@ public String getShortDetails(Context ctx) {
 		return TextUtils.join(", ", aList);
 	}
 
+	public static class MissingUuidsException extends RuntimeException {
```
Comment (werman) [inline app/src/main/java/net/programmierecke/radiodroid2/data/DataRadioStation.java]: This isn't used from what I see.

**Comment by morckx** [comment]
> Thanks!
> 
> I have one concern. What if there is a station which could not be refreshed? We then would try to refresh it until the end of times from what I see. 

So what? This is also not implemented for uuids. Should I make this special for stations with missing ids are also for those that cannot be found by their uuid?

> Also I think it would be better to check if there is an internet connection before refreshing.
> 

This would also be the first place to do that. However, this is of course needed if we have a retry to delete countdown.

I'll add something as an extra commit, although I don't think that this makes much sense.

> And a small nitpick: could you add `{` `}` to one line ifs (that's up to you but I personally debugged to many errors with ifs without braces) and definitely add braces to the for loop with more than one line.

OK, no problem.

**Comment by werman** [comment]
Thanks, 55da477 is what I meant.

**Pull request merged**